### PR TITLE
Cleanup Parquet chunked writer

### PR DESCRIPTION
**Pull request by ttnghia**
Similar to https://github.com/rapidsai/cudf/pull/13091, this changes the internal variables of Parquet chunked writer:
 * Renaming them to have a `_` prefix consistently.
 * Add `const` qualifier to some variables that are writer parameters.
 * Regroup them.

There is not any new implementation added. However, the unused parameter `mr` is removed from its interface thus this is flagged as breaking changes.

Closes:
 * https://github.com/rapidsai/cudf/issues/13079

### CODE REVIEW

File: cpp/src/io/parquet/writer_impl.hpp [hunk 1]
```diff
@@ -217,39 +213,35 @@ class writer::impl {
...
-  Compression compression_             = Compression::UNCOMPRESSED;
-  size_t max_row_group_size            = default_row_group_size_bytes;
-  size_type max_row_group_rows         = default_row_group_size_rows;
-  size_t max_page_size_bytes           = default_max_page_size_bytes;
-  size_type max_page_size_rows         = default_max_page_size_rows;
-  statistics_freq stats_granularity_   = statistics_freq::STATISTICS_NONE;
-  dictionary_policy dict_policy_       = dictionary_policy::ALWAYS;
-  size_t max_dictionary_size_          = default_max_dictionary_size;
-  bool int96_timestamps                = false;
-  int32_t column_index_truncate_length = default_column_index_truncate_length;
```
[the remark this answers is not in the document]
Comment (ttnghia) [reply cpp/src/io/parquet/writer_impl.hpp:234]: These variables are assigned from outside options passed in by user, and never changed after the writer was constructed. Thus we don't need such default values here. These default values are used for writer options construction.

[the remark this answers is not in the document]
Comment (ttnghia) [reply cpp/src/io/parquet/writer_impl.hpp:234]: See https://github.com/rapidsai/cudf/blob/branch-23.06/cpp/include/cudf/io/orc.hpp#L418-L430.

**Review by hyperbolic2346 — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Pull request merged**